packages/components/badge/src/EntityStatusBadge/EntityStatusBadge.tsx

```diff
@@ -13,20 +18,58 @@ const statusMap: { [key in EntityStatus]: BadgeVariant } = {
   new: 'primary-filled',
 };
 
-export interface EntityStatusBadgeProps
-  extends Omit<BadgeProps, 'variant' | 'children' | 'endIcon' | 'startIcon'> {
-  entityStatus: EntityStatus;
-}
+type BadgeSizeWithIsScheduledProp =
+  | {
+      /**
+       * Sets the size of the component
+       * @default default
+       */
+      size?: Exclude<BadgeSize, 'small'>;
+      /**
+       * Indicates that the entity that the badge is related to has a scheduled action
+       */
+      isScheduled?: boolean;
+    }
+  | {
+      /**
+       * Sets the size of the component
+       * @default default
+       */
+      size: PickUnion<BadgeSize, 'small'>;
+      // We use discriminative union typing, so in case the size is set as small we don't allow isScheduled
+      isScheduled?: never;
+    };
+
+export type EntityStatusBadgeProps = Omit<
+  BadgeProps,
+  'children' | 'endIcon' | 'size' | 'startIcon' | 'variant'
+> &
+  BadgeSizeWithIsScheduledProp & {
+    entityStatus: EntityStatus;
+  };
 
 function EntityStatusBadge(
   props: EntityStatusBadgeProps,
   ref: React.Ref<HTMLDivElement>,
 ) {
-  const { entityStatus, size = 'default', ...otherProps } = props;
-
+  const {
+    entityStatus,
+    isScheduled = undefined,
+    size = 'default',
+    ...otherProps
+  } = props;
+  const withIcon = isScheduled && size === 'default';
   const variant = statusMap[entityStatus];
+
   return (
-    <Badge {...otherProps} size={size} variant={variant} ref={ref}>
+    <Badge
+      {...otherProps}
+      {...(withIcon
+        ? { size, startIcon: <ClockIcon testId="schedule-icon" /> }
+        : { size })}
+      variant={variant}
+      ref={ref}
+    >
```

Comment:
nit
```suggestion
    <Badge
      {...otherProps}
      size={size}
      icon={withClockIcon ? <ClockIcon testId="schedule-icon" /> : undefined}
      variant={variant}
      ref={ref}
    >
```

Reply:
Typescript doesn't accept it because the startIcon can only be passed when `size === 'small'` 😞